Changed packages that are installed with running with the -d flag

amdkila opened the pull request:
- Removes gtest and lapack as library dependency
- Removes libgomp as a library dependency
- Removes blis as a library dependency 

All are now client dependencies

amdkila committed:
Changed packages that are installed with running with the -d flag

TorreZuk commented on install.sh:
```diff
@@ -364,8 +362,10 @@ esac
   install_packages
+fi
+
+if [[ "${build_clients}" == true ]]; then
```
Would only want this on install_dependencies case no?

leekillough commented on install.sh:
```diff
@@ -135,17 +135,15 @@ install_packages( )
   # dependencies needed for rocblas and clients to build
```
Comment is misleading: Does this part of the script add dependencies for `rocblas` _and_ `clients` to build? If so, why is `libgomp` being removed? Or is this part of the script only for `rocblas`, and not for `clients`?

amdkila replied:
This is the part of the script for building without the client. I have fixed this in my latest commit

zaliu reviewed (changes requested):
Can we please pull rocm/rocm-terminal:2.9 and build rocBLAS with this PR and `install.sh -d` to see if it succeeds?  I just tried it and it failed when building TensileHost for a missing header <boost/algorithm/string/split.hpp>.  Thanks.
Moving the libboost-program-options-dev package from client_dependencies to library_dependencies made `install.sh -d` work - at least on Ubuntu.

amdkila committed:
Made install script fixes, and added boost as a library dependency to fix missing header when building TensileHost

amdkila committed:
Added boost when only building client

amdkila commented:
Yeah, I added boost as a client dependency because I thought CI would fail without it as we only build the client and then define the dependencies in the dockerfile. However, it turns out boost was already defined in the dockerfile (which is where it should be defined for CI, not added as a client dependency). I removed that commit, and it has passed CI now (only failure was an I/O error on Ubuntu), so this should be safe to merge

leekillough commented:
rocblas client, but not the library, does depend on Boost.

It is used to parse command-line arguments for rocblas-test and rocblas-bench.

amdkila committed:
Moved boost back to client packaging

amdkila commented:
It isn't. I removed it in my last commit: https://github.com/ROCmSoftwarePlatform/rocBLAS/pull/764/commits/0abd28a054e6c080beac8d241324e44e257b4696

amcamd reviewed (approved):
LGTM

amdkila commented:
Looks like @zaliu has to remove his request for changes before I can merge the PR
